test/dshell/dwarf.d

```diff
@@ -0,0 +1,74 @@
+import dshell;
+
+import std.algorithm;
+import std.file;
+import std.string;
+import std.path;
+import std.process;
+
+int main()
+{
+    version(DigitalMars) { }
+    else
+    {
+        writeln("Skipping dwarf.d for non-DMD compilers.");
+        return DISABLED;
+    }
+
+    version(Windows)
+    {
+        writeln("Skipping dwarf.d for Windows.");
+        return DISABLED;
+    }
+
+    // test them all
+    foreach (string path; dirEntries(EXTRA_FILES ~ "/dwarf/", SpanMode.shallow))
+    {
+        if (!isFile(path) || extension(path) != ".d")
+            continue;
+
+        // retrieve the filename without the extension
+        auto filename = baseName(stripExtension(path));
+
+        string exe = OUTPUT_BASE ~ "/" ~ filename;
+        writeln("EXE: ", exe);
+        run("$DMD -m$MODEL -of" ~ exe ~ "$EXE -conf= -fPIE -g"
+            ~ " -I$EXTRA_FILES/dwarf/ " ~ path);
+
+        //import std.outbuffer;
+        //OutBuffer result = new OutBuffer();
+        //std.stdio.stdout = result;
+
+        // Execute the fix
+        run(exe);
+
+        // Write objdump result to a file
+        auto objdump = OUTPUT_BASE ~ "/" ~ filename ~ ".objdump";
+
+        // create the file
+        if (!exists(objdump))
+            std.file.write(objdump, "");
+
+        auto objdump_file = File(objdump, "w");
+        run("objdump -W " ~ exe, objdump_file);
+        objdump_file.close();
+
+        // Objdump excepted results
+        string excepted_results_file = EXTRA_FILES ~ "/dwarf/excepted_results/"
+            ~ filename ~ ".txt";
+
+        if (!exists(excepted_results_file))
+            assert(0, "DWARF tests must have a .txt file in the `excepted_results`"
+                ~ " folder which contains the DWARF dump info");
+
+        // Read file result as a string
+        auto result = cast(string)readText(objdump);
+        foreach (line; File(excepted_results_file).byLine)
+        {
+            assert(canFind(result, line), filename ~ ": Couln't find `" ~ line ~ "` in the DWARF dump info.");
```

Comment:
Nope. `objdump` is very large (~322 lines for `fix21153.d`). That's why I set `stdout` into an external file.

Reply:
Knowing which line is missing without knowing which are present makes for a horrible debug experience on the CI machines.

Do your test actually require linking? Druntime is built without debug symbols and your tests still pass when checking the object file (which causes a lot less unrelated output from `objdump`)

Reply:
Yes I need the Druntime, because I'll need to check some infos with the `main()` func (issue [21160](https://issues.dlang.org/show_bug.cgi?id=21160)).

I'll simply write the content of the file into `stdout`.